core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorPartitioningHandle.java

```diff
@@ -24,4 +24,10 @@ default boolean isCoordinatorOnly()
     {
         return false;
     }
+
+    // Specifies if writing to partition has to be performed by a single writer instance
+    default boolean isSingleWriterPerPartition()
```

Comment:
partitioning handle is insert agnostic. After some thought, it's better to put it in `ConnectorMetadata`:
```
boolean ConnectorMetadata#isSingleWriterPerPartition(ConnectorSession session, ConnectorPartitioningHandle partitioningHandle)
```